src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -37,39 +32,56 @@
 DELETE_STARTER_CHUNKS_PER_DOC = 32
 
 
-def connect_to_pinecone():
+@lru_cache(maxsize=1)
+def _get_global_client() -> Pinecone:
+    return Pinecone()
+
+
+def connect_to_pinecone(pinecone_client: Pinecone = None):
```

Comment:
Consider changing function name to "check_pinecone_client_connection" this method is not doing "init" anymore